R/mergetables.R

```diff
@@ -159,16 +158,22 @@ Merge2Tables <- function(left, right, direction = c("Side-by-side", "Up-and-down
         }
         stop("Can not find any matching ", type, ". Perhaps you meant to join ", other.direction, "?")
     }
-
-    if (any(is.na(rownames(left))) || any(is.na(rownames(right))))
-    {
-        stop("Objects to be merged must both have a unique set of names without NAs, however NAs have been found.")
-    }
-
-    if (length(unique(rownames(left))) != nrow(left) || length(unique(rownames(right))) != nrow(right))
-    {
-        stop("Objects to be merged must both have a unique set of names without NAs, however duplicate names have been found.")
-    }
+    if (any(is.na(rownames(left))))
+        stop("Row ", which(is.na(rownames(left))), " in ", left.name, " have missing names.")
+    if (any(is.na(rownames(right))))
+        stop("Row ", which(is.na(rownames(right))), " in ", right.name, " have missing names.")
```

Comment:
Add sentence to help user fix the error. i.e. "Please give the affected rows a unique name and then run merge tables again" or the equivalent set of steps to fix the error.
Optional improvement, implement handling of singular and plural cases in a way that handles automatic translation.
https://cran.r-project.org/doc/manuals/r-release/R-exts.html#Diagnostic-messages

Reply:
Fixed in commit 4c17bb3